core/libcamera_app.cpp

```diff
@@ -96,12 +99,19 @@ static void set_pipeline_configuration()
 LibcameraApp::LibcameraApp(std::unique_ptr<Options> opts)
 	: options_(std::move(opts)), controls_(controls::controls), post_processor_(this)
 {
-	check_camera_stack();
+	LibcameraApp::Platform platform = get_platform();
+	if (platform == Platform::LEGACY)
+	{
+		// If we definitely appear to be running the old camera stack, complain and give up.
+		// Everything else, Pi or not, we let through.
+		fprintf(stderr, "ERROR: the system appears to be configured for the legacy camera stack\n");
+		exit(-1);
+	}
```

Comment:
How about the UNKNOWN case? Might it be better to quit with a helpful message rather than let it run on and for some slightly unspecified failure? (I know it should never happen...)

Reply:
Done.